doc: add source community condition definition (#43376)
nodejs/node: doc/api/packages.md

* `"types"` - can be used by typing systems to resolve the typing file for
  the given export. _This condition should always be included first._

> * `"source"` - can be used to reference unminified / unoptimized source versions
>   to use for debugging workflows, that will behave identically to any non-source
>   variants.

Reviewer: ```suggestion
* `"source"` - can be used to reference unminified, unoptimized, _runnable_
  source JavaScript files. These can be used for debugging workflows, that will
  behave identically to any other variants.
```

Author: The only other word I can think of off-hand that captures the same meaning might be `"unoptimized"`.

Author: `"pretty"` is close, but it's not just unminification, it's also having separated source files structured by the author.
